core/templates/pages/topic-editor-page/editor-tab/topic-editor-tab.directive.html

```diff
@@ -1,81 +1,216 @@
+
+      <div class="topic-name">
+        <label for="topicName" class="form-heading protractor-test-topic-name-heading">Topic Name</label>
+        <input id="topicName" type="text" class="form-control protractor-test-topic-name-field"
+               ng-disabled="!topicRights.canEditTopic()"
+               ng-model="editableName" ng-blur="updateTopicName(editableName)"
+               placeholder="Enter a name for the topic." maxlength="<[MAX_CHARS_IN_TOPIC_NAME]>" ng-trim="false">
+        <span class="oppia-input-box-subtitle">
+          <i>
+            Topic name should be at most <[MAX_CHARS_IN_TOPIC_NAME]> characters.
+          </i>
+        </span>
+      </div>
+
+      <div class="topic-description" ng-class="{'has-error': editableDescriptionIsEmpty && topicDescriptionChanged}">
+        <label for="topicDescription" class="form-heading protractor-test-topic-description-heading">Description</label>
+        <textarea type="text" class="form-control protractor-test-topic-description-field"
+                  ng-model="editableDescription" maxlength="<[MAX_CHARS_IN_TOPIC_DESCRIPTION]>"
+                  ng-change="updateTopicDescriptionStatus(editableDescription)"
+                  ng-blur="updateTopicDescription(editableDescription)"
+                  placeholder="Enter the description of the topic"
+                  style="height: 10vh;">
+        </textarea>
+        <span class="oppia-input-box-subtitle">
+          <i>
+            Topic description should be at most <[MAX_CHARS_IN_TOPIC_DESCRIPTION]> characters.
+          </i>
+        </span>
+        <div>
+          <button class="btn btn-secondary" ng-click="togglePreview()">Show/Hide Preview</button>
+          <div ng-if="topic.getId() && topicPreviewCardIsShown">
+            <preview-thumbnail name="editableName"
+                               description="editableDescription"
+                               preview-footer="topic.getCanonicalStoryIds().length"
+                               filename = "topic.getThumbnailFilename()"
+                               thumbnail-bg-color="topic.getThumbnailBgColor()"
+                               bg-color="'#2F6687'">
+            </preview-thumbnail>
+          </div>
+        </div>
+        <span ng-if="editableDescriptionIsEmpty && topicDescriptionChanged" class="form-text" style="font-size: smaller">
         What does this topic contain?
-      </span>
-    </div>
-
-    <div class="canonical-stories">
-      <label class="form-heading"> Canonical Stories </label>
-      <button class="btn btn-secondary protractor-test-create-story-button"
-              ng-click="createCanonicalStory()"
-              style="float: right;">
-        Create Story
+        </span>
+      </div>
+    </md-card>
+    <md-card class="oppia-subtopics-list-card">
+      <h3>Subtopics</h3>
+      <button class="btn add-subtopic-btn protractor-test-add-subtopic-button" ng-click="createSubtopic()">
+        + ADD SUBTOPIC
```

Comment:
After I click Add Subtopic and then cancel creation, I still get warnings that all subtopics should have thumbnails, skills and so on. (Only goes away when I refresh the page)

Reply:
Ok fixed this.